src/Formatters/CharsetFormatter.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.CodingConventions;
+
+namespace Microsoft.CodeAnalysis.Tools.Formatters
+{
+    internal sealed class CharsetFormatter : DocumentFormatter
+    {
+        protected override string FormatWarningDescription => Resources.Fix_file_encoding;
+
+        private static Encoding Utf8 => new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
+        private static Encoding Latin1 => Encoding.GetEncoding("iso-8859-1");
+
+        protected override Task<SourceText> FormatFileAsync(
+            Document document,
+            SourceText sourceText,
+            OptionSet options,
+            ICodingConventionsSnapshot codingConventions,
+            FormatOptions formatOptions,
+            ILogger logger,
+            CancellationToken cancellationToken)
+        {
+            return Task.Run(() =>
+            {
+                if (!TryGetCharset(codingConventions, out var encoding)
+                    || sourceText.Encoding.Equals(encoding))
+                {
+                    return sourceText;
+                }
+
+                return SourceText.From(sourceText.ToString(), encoding);
+            });
+        }
+
+        private static bool TryGetCharset(ICodingConventionsSnapshot codingConventions, out Encoding encoding)
+        {
+            if (codingConventions.TryGetConventionValue("charset", out string charsetOption))
+            {
+                encoding = GetCharset(charsetOption);
+                return true;
+            }
+
+            encoding = null;
+            return false;
+        }
+
+        public static Encoding GetCharset(string charsetOption)
+        {
+            switch (charsetOption)
+            {
+                case "latin1":
+                    return Latin1;
+                default:
+                case "utf-8":
```

Comment:
it feels weird to have the default label mixed in with the case labels instead of being at the end.

Consider using a switch expression like the following

```cs
charsetOption switch
{
    "latin1" => Latin1,
    "utf-8-bom" => Encoding.UTF8,
    "utf-16be" => Encoding.BigEndianUnicode,
    "utf-16le" => Encoding.Unicode,
    _ => Utf8
}
```